openedx_tagging/core/tagging/models.py

```diff
@@ -558,6 +555,38 @@ def is_valid(self, check_taxonomy=True, check_tag=True, check_object=True) -> bo
 
         return True
 
+    def get_tag_options(self) -> Iterator[TagResult]:
```

Comment:
@ChrisChV 

> 👍 to this. Yes, some system defined taxonomies need to overwrite get_tags

I chose to address this here, by converting the functionality from [Taxonomy.get_tags](https://github.com/openedx/openedx-learning/pull/62/commits/321fd464bcdc4b1bc6dd39b1175180c0177b95a4#diff-204875f41064da3abfd4648611ad41c6f21abe09277263410e3daafbd343f2a3L207-L233) to this ObjectTag method.

This is the ClosedObjectTag version of this, which yields actual tags.
The [superclass version yields nothing](https://github.com/openedx/openedx-learning/pull/62/commits/321fd464bcdc4b1bc6dd39b1175180c0177b95a4#diff-204875f41064da3abfd4648611ad41c6f21abe09277263410e3daafbd343f2a3R455-R462), so OpenObjectTags have no tag options by default.

System taxonomy ObjectTag subclasses can use this to generate a dynamic list of tag options as needed.